### PR TITLE
Added PrintsPage section; added Java Examples to section

### PR DESCRIPTION
**Pull request by shbenzer**
### **User description**
Added PrintsPage section detailing the ways to print a page (browsingcontext or printspage). updated section descriptions using genai to be more technical and read better. Added changes to all translations.

### Description
Added PrintsPageTest.java
updated print_page.md in all languages.

### Motivation and Context
make site more comprehensive
hacktober
Issue #1941 

### Types of changes

- [x] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [x] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist

- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.

___

### **PR Type**
Documentation, Tests

___

### **Description**
- Added a new Java test class `PrintsPageTest` with methods to test printing functionality using `PrintsPage` and `BrowsingContext`.
- Enhanced documentation across multiple languages (English, Japanese, Portuguese, Chinese) to include detailed information on printing web pages using Selenium.
- Introduced `PrintsPage` and `BrowsingContext` classes in the documentation, along with configuration options for printing.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>PrintsPageTest.java</strong><dd><code>Add Java test cases for printing functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/java/src/test/java/dev/selenium/interactions/PrintsPageTest.java

<li>Added a new test class <code>PrintsPageTest</code>.<br> <li> Implemented two test methods for printing using <code>PrintsPage</code> and <br><code>BrowsingContext</code>.<br> <li> Utilized <code>PrintOptions</code> for configuring print settings.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/2005/files#diff-2ba4957f9ab1f581acdc5c3f723e1acb4e14a305b04592175703e95bd8531812">+31/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>print_page.en.md</strong><dd><code>Enhance English documentation for printing web pages</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/interactions/print_page.en.md

<li>Expanded documentation on printing web pages with Selenium.<br> <li> Introduced <code>PrintsPage</code> and <code>BrowsingContext</code> classes.<br> <li> Detailed configuration options for printing.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/2005/files#diff-ab83295d85363fa0e82b30ca84fb9727d0e73fecc3de7b707767c39343a2459c">+51/-10</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>print_page.ja.md</strong><dd><code>Update Japanese documentation for printing features</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/interactions/print_page.ja.md

<li>Updated Japanese documentation for printing web pages.<br> <li> Added details on <code>PrintsPage</code> and <code>BrowsingContext</code>.<br> <li> Improved configuration instructions.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/2005/files#diff-b2a566eb1a7cdfae34aa2dd15f597e4b8fd40fbdaadc1a3db3ce0fe850e7dc92">+51/-10</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>print_page.pt-br.md</strong><dd><code>Update Portuguese documentation for printing features</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/interactions/print_page.pt-br.md

<li>Updated Portuguese documentation for printing web pages.<br> <li> Included <code>PrintsPage</code> and <code>BrowsingContext</code> details.<br> <li> Enhanced configuration guidance.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/2005/files#diff-d4342199d4feb4c990d7fa275bf4912397b428908760f58d26e1326ba4aea21a">+51/-10</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>print_page.zh-cn.md</strong><dd><code>Update Chinese documentation for printing features</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/interactions/print_page.zh-cn.md

<li>Updated Chinese documentation for printing web pages.<br> <li> Added information on <code>PrintsPage</code> and <code>BrowsingContext</code>.<br> <li> Improved configuration instructions.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/2005/files#diff-33cd5d0d8234827d7cacbb49b609e33192310123a67cbe0910f68a66a2b5eb6d">+51/-10</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: website_and_docs/content/documentation/webdriver/interactions/print_page.en.md [hunk 1]
```diff
@@ -177,4 +181,41 @@ Using `getBackground()` and `setBackground()` methods, you can get/set whether t
...
+{{< /tabpane >}}
+
+## Printing
+
+Once you've configured your PrintOptions, you're ready to print the page. To do this, 
+you can invoke the print function, which generates a PDF representation of the web page. 
+The resulting PDF can be saved to your local storage for further use or distribution. 
+Using `PrintsPage()`, the print command will return the PDF data in base64-encoded format, which can be decoded 
+and written to a file in your desired location, and using `BrowsingContext()` will return a String. `BrowsingContext()`
+is part of Selenium's BiDi implementation.
```
Comment (harsha509) [inline website_and_docs/content/documentation/webdriver/interactions/print_page.en.md]: may be making a note will be useful like

**Note**:  `BrowsingContext()` is part of Selenium's BiDi implementation. To enable BiDi see "Enabling BiDi" see https://www.selenium.dev/documentation/webdriver/bidi/

Any thoughts on this @shbenzer !

Comment (shbenzer) [reply website_and_docs/content/documentation/webdriver/interactions/print_page.en.md]: I like that - I'll add it in the morning

Comment (shbenzer) [reply website_and_docs/content/documentation/webdriver/interactions/print_page.en.md]: Added

**Review by harsha509 — APPROVED** [review]
Thank you @shbenzer !

**Comment by shbenzer** [comment]
@harsha509 what should this embedded link be instead?

`[Enabling Bidi]({{< ref "../bidi/_index.md" >}}) `

**Comment by harsha509** [comment]
> @harsha509 what should this embedded link be instead?
> 
> `[Enabling Bidi]({{< ref "../bidi/_index.md" >}}) `

[Enabling Bidi]({{< ref "bidi/" >}})

**Comment by shbenzer** [comment]
> > @harsha509 what should this embedded link be instead?
> > `[Enabling Bidi]({{< ref "../bidi/_index.md" >}}) `
> 
> [Enabling Bidi]({{< ref "bidi/" >}})

No index.md?

**Comment by harsha509** [comment]
> > > @harsha509 what should this embedded link be instead?
> > > `[Enabling Bidi]({{< ref "../bidi/_index.md" >}}) `
> > 
> > 
> > [Enabling Bidi]({{< ref "bidi/" >}})
> 
> No index.md?

Not required!
it lands on index page by default!

**Comment by shbenzer** [comment]
It works! Thank you!

**Comment by shbenzer** [comment]
Should be ready to merge, thank you @harsha509

**Comment by shbenzer** [comment]
Could the issue be that there isn't a bidi/_index.md for the other 3 languages?

**Comment by harsha509** [comment]
Oh, 
I just noticed that we dont have  other translated pages for https://www.selenium.dev/documentation/webdriver/bidi/

it only works for en for now. I will add pages for bidi in other languages!

**Comment by shbenzer** [comment]
these imports are killing me lol

**Comment by shbenzer** [comment]
Do you know how to fix this error @harsha509 :

```
Error:  Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.488 s <<< FAILURE! -- in dev.selenium.interactions.PrintsPageTest
Error:  dev.selenium.interactions.PrintsPageTest.PrintWithBrowsingContextTest -- Time elapsed: 0.569 s <<< ERROR!
org.openqa.selenium.bidi.BiDiException: 
Unable to create a BiDi connection
Build info: version: '4.25.0', revision: '8a8aea2337'
System info: os.name: 'Linux', os.arch: 'amd64', os.version: '6.5.0-1025-azure', java.version: '11.0.24'
Driver info: BiDi Connection
	at org.openqa.selenium.bidi.HasBiDi.lambda$getBiDi$0(HasBiDi.java:25)
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at org.openqa.selenium.bidi.HasBiDi.getBiDi(HasBiDi.java:25)
	at org.openqa.selenium.bidi.browsingcontext.BrowsingContext.<init>(BrowsingContext.java:88)
	at dev.selenium.interactions.PrintsPageTest.PrintWithBrowsingContextTest(PrintsPageTest.java:27)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:396)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:721)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:396)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:721)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by harsha509** [comment]
> Do you know how to fix this error @harsha509 :
> 
> ```
> Error:  Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.488 s <<< FAILURE! -- in dev.selenium.interactions.PrintsPageTest
> Error:  dev.selenium.interactions.PrintsPageTest.PrintWithBrowsingContextTest -- Time elapsed: 0.569 s <<< ERROR!
> org.openqa.selenium.bidi.BiDiException: 
> Unable to create a BiDi connection
> Build info: version: '4.25.0', revision: '8a8aea2337'
> System info: os.name: 'Linux', os.arch: 'amd64', os.version: '6.5.0-1025-azure', java.version: '11.0.24'
> Driver info: BiDi Connection
> 	at org.openqa.selenium.bidi.HasBiDi.lambda$getBiDi$0(HasBiDi.java:25)
> 	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
> 	at org.openqa.selenium.bidi.HasBiDi.getBiDi(HasBiDi.java:25)
> 	at org.openqa.selenium.bidi.browsingcontext.BrowsingContext.<init>(BrowsingContext.java:88)
> 	at dev.selenium.interactions.PrintsPageTest.PrintWithBrowsingContextTest(PrintsPageTest.java:27)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
> 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
> 	at java.base/java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:396)
> 	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:721)
> 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
> 	at java.base/java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:396)
> 	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:721)
> 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
> 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
> 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
> 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
> 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
> ```

Sure let me check!

**Comment by harsha509** [comment]
Hi @shbenzer ,

Looked into the code. you need to set capability `options.setCapability("webSocketUrl", true);` to enable bidi connection.

Here is the updated example

```
package dev.selenium.interactions;

import org.openqa.selenium.Pdf;
import org.openqa.selenium.bidi.browsingcontext.BrowsingContext;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.openqa.selenium.print.PageMargin;
import org.openqa.selenium.print.PrintOptions;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.PrintsPage;
import dev.selenium.BaseTest;

public class PrintsPageTest extends BaseTest{

    @BeforeEach
    public void setup() {
        ChromeOptions options = new ChromeOptions();
        options.setCapability("webSocketUrl", true);
        driver = new ChromeDriver(options);
    }

    @Test
    public void PrintWithPrintsPageTest() 
    {
        driver.get("https://www.selenium.dev/");
        PrintsPage printer = (PrintsPage) driver;
        PrintOptions printOptions = new PrintOptions();
        Pdf printedPage = printer.print(printOptions);
        Assertions.assertNotNull(printedPage);
    }

    @Test
    public void PrintWithBrowsingContextTest() 
    {
        BrowsingContext browsingContext = new BrowsingContext(driver, driver.getWindowHandle());
        driver.get("https://www.selenium.dev/selenium/web/formPage.html");
        PrintOptions printOptions = new PrintOptions();
        String printPage = browsingContext.print(printOptions);
        Assertions.assertTrue(printPage.length() > 0);
    }
}
```

**Comment by shbenzer** [comment]
Ahh, that makes sense. Thank you!

**Comment by shbenzer** [comment]
Ahh, that makes sense. Thank you @harsha509 !

File: website_and_docs/content/documentation/webdriver/interactions/print_page.en.md [hunk 2]
```diff
@@ -177,4 +181,42 @@ Using `getBackground()` and `setBackground()` methods, you can get/set whether t
...
+There may currently be multiple implementations depending on your language of choice. For example, with Java you
+have the ability to print using either `BrowingContext()` or `PrintsPage()`. Both take `PrintOptions()` objects as a
+parameter.
+
+Note: `BrowsingContext()` is part of Selenium's BiDi implementation. To enable BiDi see [Enabling Bidi]({{< ref "bidi/" >}}) 
+
+{{< tabpane text=true >}}
+{{% tab header="Java" %}}
+**PrintsPage()**
+{{< gh-codeblock path="examples/java/src/test/java/dev/selenium/interactions/PrintsPageTest.java#L13-L20" >}}
```
Comment (harsha509) [inline website_and_docs/content/documentation/webdriver/interactions/print_page.en.md]: I believe the code lines have been modified. Could you please make the necessary adjustments @shbenzer?

**Review by harsha509 — CHANGES_REQUESTED** [review]

File: website_and_docs/content/documentation/webdriver/interactions/print_page.en.md (hunk 2 above)
Comment (shbenzer) [reply website_and_docs/content/documentation/webdriver/interactions/print_page.en.md]: Yep, done.

**Review by harsha509 — APPROVED** [review]
Thank you @shbenzer ! 🚀

**Pull request merged**